counterparty-core/counterpartycore/lib/blocks.py

```diff
@@ -107,8 +107,22 @@ def parse_tx(db, tx):
 
     try:
         with db:
+            if tx["data"] and len(tx["data"]) > 1:
+                try:
+                    message_type_id, message = message_type.unpack(tx["data"], tx["block_index"])
+                except struct.error:  # Deterministically raised.
+                    message_type_id = None
+                    message = None
+            else:
+                message_type_id = None
+                message = None
+
             if "utxos_info" in tx and tx["utxos_info"]:
-                utxo.move_assets(db, tx)
+                if not util.enabled("spend_utxo_to_detach"):
```

Comment:
This if block seems to be the opposite of the if block below and yet they seem to perform the same action?

Reply:
Yes. Before the protocol change "spend_utxo_to_detach" we always move utxo before parsing the data.
After the protocol change we move utxo before parsing only if the transaction is not an Attach or Detach, else we move utxos after the parsing.

Reply:
Ok maybe the comment on line 124 is confusing me then. also I think you can combine the if/elif into an or?

Reply:
yeah right! I re-write the comment and used one if.